src/sensors/SensorManager.cpp

```diff
@@ -36,110 +36,122 @@ namespace SlimeVR
+                            m_Sensor[i+1] = new ICM20948Sensor(i+1, secondIMUAddress, imuRotation[i+1]);
+                    }
                 }
-                else if (secondIMUAddress == 0)
-                {
-                    m_Sensor2 = new ErroneousSensor(1, SECOND_IMU);
-                }
-                else
-                {
-                    m_Logger.trace("Secondary IMU found at address 0x%02X", secondIMUAddress);
-
-#if SECOND_IMU == IMU_BNO080 || SECOND_IMU == IMU_BNO085 || SECOND_IMU == IMU_BNO086
-                    m_Sensor2 = new BNO080Sensor(1, SECOND_IMU, secondIMUAddress, SECOND_IMU_ROTATION, PIN_IMU_INT_2);
-#elif SECOND_IMU == IMU_BNO055
-                    m_Sensor2 = new BNO055Sensor(1, secondIMUAddress, SECOND_IMU_ROTATION);
-#elif SECOND_IMU == IMU_MPU9250
-                    m_Sensor2 = new MPU9250Sensor(1, secondIMUAddress, SECOND_IMU_ROTATION);
-#elif SECOND_IMU == IMU_BMI160
-                    m_Sensor2 = new BMI160Sensor(1, secondIMUAddress, SECOND_IMU_ROTATION);
-#elif SECOND_IMU == IMU_MPU6500 || SECOND_IMU == IMU_MPU6050
-                    m_Sensor2 = new MPU6050Sensor(1, SECOND_IMU, secondIMUAddress, SECOND_IMU_ROTATION);
-#elif SECOND_IMU == IMU_ICM20948
-                    m_Sensor2 = new ICM20948Sensor(1, secondIMUAddress, SECOND_IMU_ROTATION);
-#endif
+            }
+            if (foundIMU) {
+                for (uint8_t i=0; i<IMU_COUNT; i++) {
+                    swap(i);
+                    m_Sensor[i]->motionSetup();
                 }
-
-                m_Sensor2->motionSetup();
+            } else {
+                Serial.println("[ERR] I2C: Can't find I2C device on provided addresses, scanning for all I2C devices and returning");
+                I2CSCAN::scani2cports();
             }
         }
 
         void SensorManager::postSetup()
         {
-            m_Sensor1->postSetup();
-            m_Sensor2->postSetup();
+            for (uint8_t i=0; i<IMU_COUNT; i++) {
+                swap(i);
+                m_Sensor[i]->postSetup();
+            }
         }
 
         void SensorManager::update()
         {
             // Gather IMU data
-            m_Sensor1->motionLoop();
-            m_Sensor2->motionLoop();
-
-            if (!ServerConnection::isConnected())
-            {
-                return;
+            for (uint8_t i=0; i<IMU_COUNT; i++) {
+                swap(i);
+                m_Sensor[i]->motionLoop();
+                if (ServerConnection::isConnected()) {
+                    m_Sensor[i]->sendData();
+                }
             }
+        }
 
-            // Send updates
-            m_Sensor1->sendData();
-            m_Sensor2->sendData();
+        void SensorManager::swap(int id) {
+            if (INTERNAL_MUX) {
+#ifdef ESP32
+                Wire.end();
+#endif
+                Wire.begin(PIN_IMU_SDA, imuSCLPin[id/2]);
+            } else {
+                //todo
+            }
         }
```

Comment:
What does this do? It's not *super* clear what this does so maybe a tiny comment or something would help. What is this `//todo` for? I think this todo could use an explanation